suite-native/thp/src/hooks/useThpAutoconnectAlerts.tsx

```diff
@@ -0,0 +1,85 @@
+import { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { removeThpAutoconnectThunk, startThpAutoconnectThunk, thpActions } from '@suite-common/thp';
+import { selectSelectedDevice } from '@suite-common/wallet-core';
+import { useAlert } from '@suite-native/alerts';
+import { Translation } from '@suite-native/intl';
+import { useToast } from '@suite-native/toasts';
+
+export const useThpAutoconnectAlerts = () => {
+    const dispatch = useDispatch();
+    const { showAlert } = useAlert();
+    const { showToast } = useToast();
+
+    const device = useSelector(selectSelectedDevice);
+
+    const hasAutoconnectCredential =
+        device?.thp?.credentials.some(credential => credential.autoconnect) ?? false;
```

Comment:
Please create selector for this.

Reply:
Since everything is memoized, we won't react to any device change in your useCallback, only to credentials change which is better.